### ISSUE TITLE
New kconfig version creates "include" folder with C headers

### ISSUE THREAD

**Issue by afbjorklund** [afbjorklund/kconfig-frontends#1]
It would be great if these were opt-in instead, especially when not using C (or Rust).

```
include/
include/generated
include/generated/autoconf.h
include/config
include/config/tristate.conf
include/config/auto.conf
include/config/auto.conf.cmd
```

```
include/
├── config
│   ├── auto.conf
│   ├── auto.conf.cmd
│   ├── LINUX
│   └── X8664
└── generated
    ├── autoconf.h
    └── rustc_cfg
```

The old 4.11 version did not create this directory or these files, but the 4.19 (or 6.2) does.

**Comment by afbjorklund** [comment]
Related to renaming `silentoldconfig` to `syncconfig`, perhaps ?

* https://github.com/gregkh/linux/commit/cedd55d49dee941d0e4fabb2f2f555d50c53fad7

And related changes in the build system, to get rid of `-include`

* https://github.com/gregkh/linux/commit/00c864f8903dd357471e8ee48f4d57aaa9a1a0de

-----

The "include/config/auto.conf" was introduced already in `v2.6.18`:

* https://github.com/gregkh/linux/commit/c955ccafc38e77312b4c65e5a70960080fb8a3f2

But the generation was moved from make to Kconfig, for `v4.19.0`.

Later versions also added a bunch of compiler checks and similar...

**Comment by afbjorklund** [comment]
All files are created/written by the call to `conf_write_autoconf`

```C
int conf_write_autoconf(int overwrite);
```

So adding an environment variable for this, should be enough.

`libs/parser/confdata.c`

----

docs/kconfig.txt

```
Environment variables for 'syncconfig'

KCONFIG_NOSILENTUPDATE
--------------------------------------------------
If this variable has a non-blank value, it prevents silent kernel
config updates (requires explicit updates).

KCONFIG_AUTOCONFIG
--------------------------------------------------
This environment variable can be set to specify the path & name of the
"auto.conf" file.  Its default value is "include/config/auto.conf".

KCONFIG_TRISTATE
--------------------------------------------------
This environment variable can be set to specify the path & name of the
"tristate.conf" file.  Its default value is "include/config/tristate.conf".

KCONFIG_AUTOHEADER
--------------------------------------------------
This environment variable can be set to specify the path & name of the
"autoconf.h" (header) file.
Its default value is "include/generated/autoconf.h".

```

**Comment by afbjorklund** [comment]
Fixed, ff1597bdbe6e52b65ef8706f0bb6719b7d2a73f4

You can set `KCONFIG_SYNCCONFIG`, to get the kernel behaviour back.

Or call it `kconfig-conf --syncconfig` , then the files will be created.
